src/link/Elf.zig

```diff
@@ -4565,6 +4565,16 @@ fn writeAtoms(self: *Elf) !void {
         try self.base.file.?.pwriteAll(buffer, sh_offset);
     }
 
+    for (self.thunks.items) |th| {
+        const shdr = self.shdrs.items[th.output_section_index];
+        const offset = th.value + shdr.sh_offset;
+        const buffer = try gpa.alloc(u8, th.size(self));
+        defer gpa.free(buffer);
+        var stream = std.io.fixedBufferStream(buffer);
+        try th.write(self, stream.writer());
+        try self.base.file.?.pwriteAll(buffer, offset);
```

Comment:
why are you buffering the file into memory? if you write directly to the file then it would be advantageous for many reasons.

- less memory usage
- ability to use copy_file_range
- faster

Use [fs.File.writeFileAll](https://ziglang.org/documentation/master/std/#std.fs.File.writeFileAll)

Does this pattern appear anywhere else?

Reply:
`writeFileAll` doesn't apply here as we simply write linker-generated thunks into the file. We do not copy anything over.

Reply:
`writeFileAll` might work for copying sections over however, although I would need to see a benchmark to be 100% it's more efficient due to the fact we need to resolve relocations, and not just copy the bytes over. Anyhow, what confuses me is how code that is not run during behavior tests is causing the compiler to exceed the memory usage.